src/sonic-bgpcfgd/bgpcfgd/managers_device_global.py

```diff
@@ -0,0 +1,106 @@
+from .manager import Manager
+from .log import log_err, log_debug, log_notice
+import re
+
+class DeviceGlobalCfgMgr(Manager):
+    """This class responds to change in device-specific state"""
+
+    def __init__(self, common_objs, db, table):
+        """
+        Initialize the object
+        :param common_objs: common object dictionary
+        :param db: name of the db
+        :param table: name of the table in the db
+        """
+        self.directory = common_objs['directory']
+        self.cfg_mgr = common_objs['cfg_mgr']
+        self.constants = common_objs['constants']
+        self.tsa_template = common_objs['tf'].from_file("bgpd/tsa/bgpd.tsa.isolate.conf.j2")
+        self.tsb_template = common_objs['tf'].from_file("bgpd/tsa/bgpd.tsa.unisolate.conf.j2")
+        super(DeviceGlobalCfgMgr, self).__init__(
+            common_objs,
+            [],
+            db,
+            table,
+        )
+
+    def set_handler(self, key, data):
+        log_debug("DeviceGlobalCfgMgr:: set handler")
+        """ Handle device tsa_enabled state change """
+        if not data:
+            log_err("DeviceGlobalCfgMgr:: data is None")
+            return False
+
+        if "tsa_enabled" in data:
+            self.cfg_mgr.commit()
+            self.cfg_mgr.update()
+            self.isolate_unisolate_device(data["tsa_enabled"])
+            self.directory.put(self.db_name, self.table_name, "tsa_enabled", data["tsa_enabled"])
+            return True        
+        return False
+
+    def del_handler(self, key):
+        log_debug("DeviceGlobalCfgMgr:: del handler")
+        return True
+
+    def check_state_and_get_tsa_routemaps(self, cfg):
+        """ API to get TSA route-maps if device is isolated"""
+        cmd = ""
+        if self.directory.path_exist("CONFIG_DB", "BGP_DEVICE_GLOBAL", "tsa_enabled"):
+            tsa_status = self.directory.get_slot("CONFIG_DB", "BGP_DEVICE_GLOBAL")["tsa_enabled"]
+            if tsa_status == "true":
+                cmds = cfg.replace("#012", "\n").split("\n")
+                log_notice("DeviceGlobalCfgMgr:: Device is isolated. Applying TSA route-maps")
+                cmd = self.get_tsa_routemaps(cmds)            
+        return cmd
+
+    def isolate_unisolate_device(self, tsa_status):
+        """ API to get TSA/TSB route-maps and apply configuration"""
+        cmd = "\n"
+        if tsa_status == "true":
+            log_notice("DeviceGlobalCfgMgr:: Device isolated. Executing TSA")
+            cmd += self.get_tsa_routemaps(self.cfg_mgr.get_text())
+        else:
+            log_notice("DeviceGlobalCfgMgr:: Device un-isolated. Executing TSB")
+            cmd += self.get_tsb_routemaps(self.cfg_mgr.get_text())
+
+        self.cfg_mgr.push(cmd)
+        log_debug("DeviceGlobalCfgMgr::Done")
+
+    def get_tsa_routemaps(self, cmds):
```

Comment:
get_tsa_routemaps is very similar with get_tsb_routemaps, could we combine these two function with a variable for different template?
something like 
get_ts_routemaps(self, cmds, template):